CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+# Contributing
+
+## Prerequisites
+
+Nix and direnv (recommended) are used to provide all necessary developments tools: cabal, GHC, formatters, etc.
+
+### Nix
+
+Install [Nix](https://nixos.org/nix/) (recommended).
+Then [activate Flakes support for Nix](https://nixos.wiki/wiki/Flakes) (mandatory).
+
+The nix setup ([`flake.nix`](./flake.nix) and everything under [`cells`](./cells/)) in this repo use the [Standard framework](https://github.com/divnix/std) (`std`), please refer to its [documentation](https://std.divnix.com/) for general maintenance.
+
+We also use [Haskell.nix](https://github.com/input-output-hk/haskell.nix) for building haskell packages with nix. See [`haskell.nix`](./cells/prelude/haskell.nix) configuration file and the [Haskell.nix documentation](https://input-output-hk.github.io/haskell.nix/) for maintenance.
+
+### Development environment: direnv or nix develop.
+
+Development environments are provided for the following compilers:
+
+- GHC 8.10.x (default)
+- GHC 9.2.x
+
+Upon entering a development environment for the first time, `nix` may ask you to activate the necessary option and the IOG nix cache (see `nixConfig` attribute in [`flake.nix`](./flake.nix)).
+
+#### Direnv
+
+[direnv](https://direnv.net/) is recommended, [install it for your OS and hook it to your shell](https://direnv.net/docs/installation.html).
+
+Then
+
+```console
+$ direnv allow
+```
+
+##### Use alternative compiler
+
+Check available [extra-compilers](./cells/prelude/haskell.nix#L6) (eg. `ghc924`) then:
+
+```console
+$ export DEVSHELL_TARGET=ghc924
+$ direnv reload
+```
+
+or permanently:
+
+```console
+$ echo DEVSHELL_TARGET=ghc924 >> .envrc.local
+$ direnv reload
+```
+
+#### Nix develop
+
+use `nix develop` or `nix develop .#ghc924` to enter a development shell.
+
+## Development workflows
+
+#### Available tools
+
+Available tools (`cabal`, `ghcid`, `hlint`, `weeder`, etc.) are displayed upon enterring the development shell. Type `menu` to see them again.
+
+Adding or modifying the list of development tools can be done from [devshellProfiles.nix](./cells/prelude/devshellProfiles.nix).
+
+#### Build and test
+
+using cabal:
+
+```console
+$ cabal update
+$ cabal build cardano-prelude:lib:cardano-prelude
+$ cabal run cardano-prelude-test:test:prelude-tests
+```
+
+or `std` (nix):
+
+```
+$ source <(std _carapace) # Optional, for completion.
+$ std //prelude/packages/cardano-prelude:lib:cardano-prelude:build
+$ std //prelude/apps/cardano-prelude-test:test:prelude-tests:run
+```
+
+(the std/nix build will use default compiler).
+
+#### Commits
```

Comment:
missing editorconfig? Is that checked?

Reply:
it is not checked, some editor may use it for tailor some formatting for this project, only `treefmt` not doing any change is what is checked (for now only by lefthook, should be added to CI too).